### PR TITLE
Master dockerfile

### PR DESCRIPTION
**Pull request by KyleRoarty**
Uses docker much better than before, clean up documentation, add tests for building gem5, using hipcc, running square on APU model

### CODE REVIEW

File: Dockerfile
Comment (mattsinc) [inline Dockerfile]: Based on some prior conversations with Tony, I *think* ROCm 1.6.1 is the version we should stick with for now, as it seems to work best for a wide range of apps.  But if you found that ROCm 1.6.2 worked for everything you tested, I'm fine with using that instead.

File: README.md
Comment (mattsinc) [inline README.md]: See above comment

File: README.md
Comment (mattsinc) [inline README.md]: This part is fine for now.  In the future as we add more apps, we may want to just say something like "this is the newest version of ROCm that worked with everything"

File: tests/run_tests.sh
Comment (mattsinc) [inline tests/run_tests.sh]: What does 134 mean?

**Review by mattsinc — APPROVED** [review]
I will leave it up to you if you want to make the version change before merging.

File: tests/run_tests.sh
Comment (KyleRoarty) [reply tests/run_tests.sh]: 134 is the general "Program abort" return value. It's what is returned after running square on a version of ROCm where the simulator crashes. Given that I found ROCm versions where the simulator doesn't crash, I can remove that code. It was only put in place to get dockerhub build to work.

File: gem5.patch
Comment (mattsinc) [inline gem5.patch:24]: You should double-check that this works for you in your setup, but as far as I know it should work in any system.

**Review by mattsinc — APPROVED** [review]

**Review by mattsinc — APPROVED** [review]

File: gem5.patch
Comment (KyleRoarty) [reply gem5.patch:24]: I checked and it passes in /root in the container. I re-ran the DNNMark test in the ML branch and it ran to completion, so that still works.

**Pull request merged**
